js/builder/builder_graph.js

```diff
@@ -183,7 +183,7 @@ class ItemInputNode extends InputNode {
 
             for (const [i, x] of zip2(equipment_inputs, replace_items)) { setValue(i, x); }
 
-            for (const node of item_nodes) { 
+            for (const node of equip_inputs) { 
```

Comment:
the bug here was, morph- input was only requesting an update on the item "output node" (which is responsible for combining powder input and item name input) instead of the actual item input (text change node). so it would never update